### PR TITLE
[Doc] consolidate `rect` documentation

### LINKED ISSUE #22819: [Doc]: Make rect argument consistent in the docstrings
### Documentation Link

_No response_

### Problem

The `rect` argument to functions is inconsistently documented in the codebase. We should consolidate the argument into a single choice.

The options I found searching for `rect : ` are:
```bash
rect : tuple[float, float, float, float], optional
rect : sequence of float
rect : tuple (left, bottom, right, top), default: (0, 0, 1, 1)
rect : tuple of 4 floats, default: (0, 0, 1, 1)
rect : [left, bottom, width, height]
rect : (float, float, float, float)
```

### Suggested improvement

Update all `rect` arguments to a single choice of one of the above for consistency.

### PR DESCRIPTION
**Pull request by Sync271**
## PR Summary

Consolidate `rect` documentation. Closes #22819 

## PR Checklist

**Tests and Styling**
- [ ] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [ ] New features are documented, with examples if plot related.
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

**Comment by oscargus** [comment]
I believe that there are two cases here actually:
```
rect :  tuple (left, bottom, right, top)
```
and
```
rect :  tuple (left, bottom, width, height)
```
I would personally prefer this style (possibly with default parameter when available). Also, I think that (parts of) the explanation should be kept.

**Comment by story645** [comment]
hi @Sync271, do you have time to resolve the conflict and finish up this pr?

**Comment by Sync271** [comment]
> hi @Sync271, do you have time/interest to resolve the merge conflict and finish up this pr?

Will do!

File: lib/matplotlib/layout_engine.py [hunk 1]
```diff
@@ -123,10 +123,9 @@ def __init__(self, *, pad=1.08, h_pad=None, w_pad=None,
         h_pad, w_pad : float
             Padding (height/width) between edges of adjacent subplots.
             Defaults to *pad*.
-        rect : tuple of 4 floats, optional
-            (left, bottom, right, top) rectangle in normalized figure
-            coordinates that the subplots (including labels)
-            will fit into. Defaults to using the entire figure.
+        rect : tuple (left, bottom, right, top), default: (0, 0, 1, 1).
+            rectangle in normalized figure coordinates that the subplots (including labels)
+            will fit into.
```
Comment (jklymak) [inline lib/matplotlib/layout_engine.py:128]:
```suggestion
            rectangle in normalized figure coordinates that the subplots 
            (including labels) will fit into.
```

File: lib/matplotlib/gridspec.py [hunk 2]
```diff
@@ -460,7 +460,7 @@ def tight_layout(self, figure, renderer=None,
         h_pad, w_pad : float, optional
             Padding (height/width) between edges of adjacent subplots.
             Defaults to *pad*.
-        rect : tuple of 4 floats, default: (0, 0, 1, 1), i.e. the whole figure
+        rect : tuple (left, bottom, right, top), default: None. i.e. the whole figure
```
Comment (jklymak) [inline lib/matplotlib/gridspec.py:463]:
```suggestion
        rect : tuple (left, bottom, right, top), default: None.
```

File: lib/matplotlib/gridspec.py [hunk 3]
```diff
@@ -460,7 +460,7 @@ def tight_layout(self, figure, renderer=None,
...
             (left, bottom, right, top) rectangle in normalized figure
             coordinates that the whole subplots area (including labels) will
             fit into.
```
Comment (jklymak) [inline lib/matplotlib/gridspec.py:466]:
```suggestion
            fit into.  Default (None) is the whole figure.
```

File: lib/matplotlib/figure.py [hunk 4]
```diff
@@ -519,8 +519,8 @@ def add_axes(self, *args, **kwargs):
 
         Parameters
         ----------
-        rect : sequence of float
-            The dimensions [left, bottom, width, height] of the new Axes. All
+        rect : tuple (left, bottom, right, top)
+            The dimensions (left, bottom, right, top) of the new Axes. All
```
Comment (jklymak) [inline lib/matplotlib/figure.py:523]: Pretty sure this change is incorrect.

**Review by jklymak — CHANGES_REQUESTED** [review]

File: lib/matplotlib/gridspec.py (hunk 2 above)
Comment (QuLogic) [reply lib/matplotlib/gridspec.py:463]: Leave out the trailing `.` as well.

File: lib/matplotlib/figure.py (hunk 4 above)
Comment (Sync271) [reply lib/matplotlib/figure.py:523]: What would be the correct way to put `[x_min, y_min, dx, dy]`?

**Review by jklymak — APPROVED** [review]

**Comment by jklymak** [comment]
Thanks a lot @Sync271 - I squashed your commits into one, which is our usual policy...

**Pull request merged**